velox/functions/prestosql/window/Rank.cpp

```diff
@@ -41,17 +51,49 @@ class RankFunction : public exec::WindowFunction {
       vector_size_t resultOffset,
       const VectorPtr& result) override {
     int numRows = peerGroupStarts->size() / sizeof(vector_size_t);
-    auto* rawValues = result->asFlatVector<int64_t>()->mutableRawValues();
     auto* rawPeerStarts = peerGroupStarts->as<vector_size_t>();
+
+    void* rawValues = nullptr;
+    switch (type) {
+      case RankType::kRank:
+      case RankType::kDenseRank: {
+        rawValues = result->asFlatVector<int64_t>()->mutableRawValues();
+        break;
+      }
+      case RankType::kPercentRank: {
+        rawValues = result->asFlatVector<double>()->mutableRawValues();
+        break;
+      }
+    }
+
     for (int i = 0; i < numRows; i++) {
       auto start = rawPeerStarts[i];
       if (start != currentPeerGroupStart_) {
         currentPeerGroupStart_ = start;
-        rank_ += previousPeerCount_;
+        if constexpr (type == RankType::kDenseRank) {
+          rank_ += (previousPeerCount_ != 0);
+        } else {
+          rank_ += previousPeerCount_;
+        }
         previousPeerCount_ = 0;
       }
 
-      rawValues[resultOffset + i] = rank_;
+      switch (type) {
```

Comment:
Switching on every row is expensive. Consider turning this class into a template.

Reply:
Thanks for the suggestion but the class is already templated on RankType, shall I use ``if constexpr`` here instead of ``switch``?